pkg/solana/chainwriter/transform_registry.go

```diff
@@ -2,49 +2,89 @@ package chainwriter
 
 import (
 	"context"
-	"errors"
 	"fmt"
 
 	"github.com/gagliardetto/solana-go"
 	"github.com/mitchellh/mapstructure"
-	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_offramp"
+
+	txmutils "github.com/smartcontractkit/chainlink-solana/pkg/solana/txm/utils"
+
 	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 )
 
-type ReportPostTransform struct {
-	ReportContext  [2][32]byte
-	Report         []byte
-	Info           ccipocr3.ExecuteReportInfo
-	AbstractReport ccip_offramp.ExecutionReportSingleChain
-	TokenIndexes   []byte
+type ExtraDataDecoded struct {
+	// ExtraArgsDecoded contain message specific extra args.
+	ExtraArgsDecoded map[string]any
+	// DestExecDataDecoded contain token transfer specific extra args.
+	DestExecDataDecoded []map[string]any
+}
+
+type SVMExecCallArgs struct {
+	ReportContext [2][32]byte                `mapstructure:"ReportContext"`
+	Report        []byte                     `mapstructure:"Report"`
+	Info          ccipocr3.ExecuteReportInfo `mapstructure:"Info"`
+	ExtraData     ExtraDataDecoded           `mapstructure:"ExtraData"`
+	TokenIndexes  []byte                     `mapstructure:"TokenIndexes"`
+}
+
+type SVMCommitCallArgs struct {
+	ReportContext [2][32]byte               `mapstructure:"ReportContext"`
+	Report        []byte                    `mapstructure:"Report"`
+	Rs            [][32]byte                `mapstructure:"Rs"`
+	Ss            [][32]byte                `mapstructure:"Ss"`
+	RawVs         [32]byte                  `mapstructure:"RawVs"`
+	Info          ccipocr3.CommitReportInfo `mapstructure:"Info"`
 }
 
-func FindTransform(id string) (func(context.Context, any, solana.AccountMetaSlice, map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, error), error) {
+// TODO: replace with actual value from CCIP on-chain
```

Comment:
Do we know when we'd get the actual value for this? Or are we moving forward with this overhead

Reply:
@matYang mentioend 120,000 would be a reasonable estimate. I am not sure whether we are expecting a more exact value cc: @agusaldasoro

Reply:
Ok if we end up moving forward with this, think we can drop the TODO.